File: src/jk_5/nailed/players/PlayerRegistry.java
package jk_5.nailed.players;

import com.google.common.collect.Maps;
import jk_5.nailed.Nailed;

import java.util.Collection;
import java.util.Map;

/**
 * No description given
 *
 * @author jk-5
 */
public class PlayerRegistry {

    private Map<String, Player> players = Maps.newHashMap();

    public Player getPlayer(String username) {
        return this.players.get(username);
    }

    public Collection<Player> getPlayers() {
        return this.players.values();
    }

    public Player getOrCreatePlayer(String username) {
        if (this.players.containsKey(username)) return this.players.get(username);
        else {
            final Player p = new Player(username);
            this.players.put(username, p);
            p.setCurrentMap(Nailed.mapLoader.getLobby());
            return p;
        }
    }
}

File: src/jk_5/nailed/players/PlayerRegistry.scala
package jk_5.nailed.players

import scala.collection.mutable
import net.minecraft.src.ICommandSender
import jk_5.nailed.map.MapLoader

/**
 * No description given
 *
 * @author jk-5
 */
object PlayerRegistry {

  private final val players = new mutable.HashMap[String, Player]()

  def getPlayer(sender: ICommandSender): Option[Player] = this.getPlayer(sender.getCommandSenderName)
  def getPlayer(username: String): Option[Player] = this.players.get(username)
  def getPlayers = this.players.values
  def getOrCreatePlayer(username: String): Player =
    if(this.getPlayer(username).isDefined) this.getPlayer(username).get
    else{
      val player = new Player(username)
      this.players.put(username, player)
      player.setCurrentMap(MapLoader.getLobby)
      player
    }
}
